include/zephyr/arch/arm/cortex_m/scripts/linker.ld

```diff
@@ -422,6 +427,11 @@ GROUP_START(DTCM)
 	SECTION_PROLOGUE(_DTCM_DATA_SECTION_NAME,,SUBALIGN(4))
 	{
 		__dtcm_data_start = .;
+
+/* Located in generated directory. This file is populated by the
+ * zephyr_linker_sources() Cmake function. */
+#include <snippets-dtcm-section.ld>
+
```

Comment:
for consistency with other includes, then please move below the `*(.dtcm_data)` and similar patterns.

Similar to:
https://github.com/zephyrproject-rtos/zephyr/blob/c3e8b731eff5beb5b28aef248bbc8ed20a2c7419/include/zephyr/arch/arm/cortex_m/scripts/linker.ld#L194-L201

same comment applies to other locations as well.